internal/controller/kyma/controller.go

```diff
@@ -209,6 +207,42 @@ func (r *Reconciler) reconcile(ctx context.Context, kyma *v1beta2.Kyma) (ctrl.Re
 	return res, err
 }
 
+func (r *Reconciler) replaceKymaFromRemote(ctx context.Context, kyma *v1beta2.Kyma) ReconciliationLoopResult {
+	updateRequired, err := r.SyncRemoteCrds.Execute(ctx, kyma)
+	if err != nil {
+		r.Metrics.RecordRequeueReason(metrics.CrdsSync, queue.UnexpectedRequeue)
+		return newReconcileResult(r.requeueWithError(ctx, kyma, fmt.Errorf("could not sync CRDs: %w", err)))
+	}
+	if updateRequired {
+		if err := r.Update(ctx, kyma); err != nil {
+			r.Metrics.RecordRequeueReason(metrics.CrdAnnotationsUpdate, queue.UnexpectedRequeue)
+			return newReconcileResult(r.requeueWithError(ctx, kyma, fmt.Errorf("could not update kyma annotations: %w", err)))
+		}
+		r.Metrics.RecordRequeueReason(metrics.CrdAnnotationsUpdate, queue.IntendedRequeue)
+		return newReconcileResult(ctrl.Result{Requeue: true}, nil)
+	}
+	// update the control-plane kyma with the changes to the spec of the remote Kyma
+	if err = r.replaceSpecFromRemote(ctx, kyma); err != nil {
+		if errors.Is(err, ErrInvalidKymaSpec) {
+			// if the spec is invalid, we should not continue with processing, but make the user aware of the issue
+			if sErr := r.updateStatusWithError(ctx, kyma, err); sErr != nil {
+				r.Metrics.RecordRequeueReason(metrics.StatusSyncToRemote, queue.UnexpectedRequeue)
+				return newReconcileResult(r.requeueWithError(ctx, kyma, fmt.Errorf("could not synchronize remote kyma status: %w", sErr)))
+			}
+			if sErr := r.syncStatusToRemote(ctx, kyma); sErr != nil {
+				r.Metrics.RecordRequeueReason(metrics.StatusSyncToRemote, queue.UnexpectedRequeue)
+				return newReconcileResult(r.requeueWithError(ctx, kyma, fmt.Errorf("could not synchronize remote kyma status: %w", sErr)))
+			}
+			return newReconcileResult(ctrl.Result{}, nil)
```

Comment:
if you return `ctrl.Result{}, nil`, then this kyma will not reconcile anymore

Reply:
update: I don't think you need to change this existing logic https://github.com/kyma-project/lifecycle-manager/pull/1748#discussion_r1711371963